sklearn/semi_supervised/label_propagation.py

```diff
@@ -264,12 +259,15 @@ def fit(self, X, y):
 
         l_previous = np.zeros((self.X_.shape[0], n_classes))
 
-        remaining_iter = self.max_iter
         unlabeled = unlabeled[:, np.newaxis]
         if sparse.isspmatrix(graph_matrix):
             graph_matrix = graph_matrix.tocsr()
-        while (_not_converged(self.label_distributions_, l_previous, self.tol)
-               and remaining_iter > 1):
+
+        self.n_iter_ = 0
+        while self.n_iter_ < self.max_iter:
```

Comment:
First, I didn't know that we could use `self.n_iter_` in the `for` loop; today I learned. :)

Second, this is the best implementation I can think of which has the same behavior as the original `while` loop:

```python
for self.n_iter_ in range(max_iter):
    # ...
else:
    # ...
    self.n_iter_ += 1
```

The `self.n_iter_ += 1` in the `else:` clause is necessary to ensure that `self.n_iter_ == self.max_iter` is possible and that it correctly happens if the loop doesn't `break` out due to convergence. This was the issue we were out to fix originally. :)

With this implementation, I am quite happy with the `for` loop as well. I'll make this change. :+1: